bay/containers/formation.py

```diff
@@ -81,7 +81,12 @@ def add_container(self, container, host):
                     break
             else:
                 # OK, we need to make one
-                instance = self.add_container(dependency, host)
+                try:
+                    instance = self.add_container(dependency, host)
+                except ImageNotFoundException as e:
+                    # Annotate the error with the container
+                    e.container = dependency
+                    raise
```

Comment:
Well, I figured it was up to the CLI code above to deal with it - this code just adds the container to the existing exception and re-raises it on up the stack so it can be handled better.